### ISSUE TITLE
`re_components` -> `re_types` migration plan

### ISSUE THREAD

**Issue by teh-cmc** [rerun-io/rerun#2679]
We want to get to the point where we can replace old components/datatypes with the new ones end-to-end, i.e. SDKs and viewer alike.

Rough outline of the plan below.

1. Implement `re_types::Loggable` for all legacy types
This should be straightforward since the new trait is much more simple than the old ones (which inherit from `arrow2-convert`) and should be implementable simply by delegating to the old one (or at least delegating to `arrow2-convert` directly).

2. Modify `re_viewer`/`re_query`/etc(?) to use the new `Loggable` trait instead of the legacy `Component`/`SerializableComponent`/`DeserializableComponent` traits
Once this is done, we can start bringing in new components one-by-one.

3. Replace legacy components with new ones
Pick a set of legacy components and **replace** them with their new counterparts _all the way_, that way there's no need to worry about binary representations at all.
We can start with all the components that make up the `Points2D` archetype, since they should already be ready.

**Comment by jleibs** [comment]
1 / 2 are done.  And I believe 3 is superseded by [#2795](https://github.com/rerun-io/rerun/issues/2795)
